### DISCUSSION TITLE
Adding prepareForValidation method

### DISCUSSION THREAD

**Discussion by fusigabs** [spatie/laravel-data#14] [Ideas]
I wonder if it would be useful to have a `prepareForValidation` method, as we currently have in `FormRequest`. This would be useful if we want to do some transformation on the request before validating and setting it in the Data class.

**Comment by rubenvanassche** [comment]
Hi @nfunwigabga,

This looks like a good idea, I would accept a documented and tested PR with this feature!

**Comment by didac-adria** [comment]
+ 1 to the feature!

**Comment by ClintWinter** [comment]
Working on it! https://github.com/spatie/laravel-data/pull/247

Please feel free to help out 😄

**Comment by Nir-An** [comment]
Hi @rubenvanassche 

where is the documentation for [modifying properties before the data pipeline](https://github.com/spatie/laravel-data/pull/247/files#diff-b4c931b0cca50f2922d28441675289a52f495fc860a305845987e87cd05a2c58)? 

I don't see it on the documentation pages 

I made a [PR 277](https://github.com/spatie/laravel-data/pull/277) for the missing documentation.

**Comment by rubenvanassche** [comment]
Looks like the docs site isn't working as it should: https://github.com/spatie/laravel-data/blob/8b4d79b53f3e9c013b52a802e02988702d3f0f72/docs/advanced-usage/pipeline.md#L67

**Reply by rubenvanassche** [reply]
Manual regenerate fixed the problem!
